Scourge Spawn

zumbak04 opened the pull request:
<!--
A basic changelog, goes to patch notes of the next version, so should be as short and informative as possible.
-->
## Changelog:
- Added this thing.
- Changed that thing.

<!--
A changelog where you can describe more complicated things to other developers and testers.
-->
## Developer changelog:
- Added this thing.
- Changed that thing.

<!--
Before the merge, we recommend you to do these tests with your branch.
-->
## Tests:
- [ ] There are no errors in `wc` files in `Documents\Paradox Interactive\Crusader Kings III\logs\error.log` except `portrait_decals.cpp:101`
- [ ] The mod takes less than 5.5 GB in the Task Manager (Windows)

<!--
If you need to explain something to testers. Otherwise, delete this part.
-->
# How to test:

LordSilvermane commented on localization/english/event_localization/wc_scourge_invasion_l_english.yml:
```diff
@@ -0,0 +1,7 @@
+﻿l_english:
+ WCCSC_5_TITLE:0 "Frozen Throne"
+ WCCSC_5_DESC:0 "Darkness has fallen upon Northrend, this day, as a mass of unearthly ice has descended from the heavens to bury itself deep within Icecrown Glacier. Those who beheld it with their own eyes have named it the Frozen Throne, for the great spire looms over all and casts a dread shadow over its new domain.\n\nFrantic tales of creatures losing their minds to madness and of the dead ris ...
```
```suggestion
 WCCSC_5_DESC:0 "Darkness has fallen upon Northrend this day, as a mass of unearthly ice has descended from the heavens to bury itself deep within Icecrown Glacier. Those who beheld it with their own eyes have named it the Frozen Throne, for the great spire looms over all and casts a dread shadow over its new domain.\n\nFrantic tales of creatures losing their minds to madness and of the dead rising as bloodthirsty monsters have quickly spread through your realm, inciting panic and despair. A terrible blight has gripped the roof of the world, and it falls to you to protect your people from this unholy scourge."
```

LordSilvermane commented on localization/english/wc_traits_l_english.yml:
```diff
@@ -489,4 +491,13 @@
+ TRAIT_FLAG_DESC_makes_incapable:0 "@warning_icon!#X Makes [wc_incapable|E]#!"
+ 
+ trait_lich_queen:0 "Lich Queen"
+ trait_lich_king_common_desc:0 "Originally created by Kil'jaeden from the spirit of the orc shaman Ner'zhul trapped inside the Helm of Domination. Lich King's unimaginable necromantic potential allows him to control hundreds of thousands of walking deads."
```
```suggestion
 trait_lich_king_common_desc:0 "Originally created by Kil'jaeden from the spirit of the orc shaman Ner'zhul trapped inside the Helm of Domination. The Lich King's unimaginable necromantic potential allows him to control hundreds of thousands of walking undead."
```

LordSilvermane commented on localization/english/wc_traits_l_english.yml:
```diff
@@ -489,4 +491,13 @@
+ trait_lich_king_common_desc:0 "Originally created by Kil'jaeden from the spirit of the orc shaman Ner'zhul trapped inside the Helm of Domination. Lich King's unimaginable necromantic potential allows him to control hundreds of thousands of walking deads."
+ 
+ trait_in_ice_prison:0 "In Ice Prison"
+ trait_in_ice_prison_desc:0 "A soul has been bounded to the Helm of Domination, seated and trapped inside a frozen block of ice which would fittingly be named the Frozen Throne."
```
```suggestion
 trait_in_ice_prison_desc:0 "A soul has been bound to the Helm of Domination, seated and trapped inside a frozen block of ice which would fittingly be named the Frozen Throne."
```

LordSilvermane commented on localization/english/wc_traits_l_english.yml:
```diff
@@ -489,4 +491,13 @@
+ trait_lich_queen:0 "Lich Queen"
+ trait_lich_king_common_desc:0 "Originally created by Kil'jaeden from the spirit of the orc shaman Ner'zhul trapped inside the Helm of Domination. Lich King's unimaginable necromantic potential allows him to control hundreds of thousands of walking deads."
+ 
+ trait_in_ice_prison:0 "In Ice Prison"
```
In Ice Prison just doesn't sound right. Could also be changed to something like 'Frozen Throne' or 'Imprisoned in Icecrown', alternatively.
```suggestion
 trait_in_ice_prison:0 "Inside an Ice Prison"
```

zumbak04 replied:
It's a title, so should be without articles.
```suggestion
 trait_in_ice_prison:0 "Inside Ice Prison"
```

LordSilvermane commented on localization/english/wc_traits_l_english.yml:
```diff
@@ -489,4 +491,13 @@
+ 
+ trait_in_ice_prison:0 "In Ice Prison"
+ trait_in_ice_prison_desc:0 "A soul has been bounded to the Helm of Domination, seated and trapped inside a frozen block of ice which would fittingly be named the Frozen Throne."
+ trait_in_ice_prison_character_desc:0 "[ROOT.GetCharacter.GetFirstNamePossessiveNoTooltip] soul has been bounded to the Helm of Domination, seated and trapped inside a frozen block of ice which would fittingly be named the Frozen Throne."
```
```suggestion
 trait_in_ice_prison_character_desc:0 "[ROOT.GetCharacter.GetFirstNamePossessiveNoTooltip] soul has been bound to the Helm of Domination, seated and trapped inside a frozen block of ice which would fittingly be named the Frozen Throne."
```

LordSilvermane reviewed (commented):
Few suggestions and corrections.

zumbak04 commented on localization/english/wc_character_interactions_l_english.yml:
```diff
@@ -1,2 +1,9 @@
\ No newline at end of file
+ give_magic_eyes_interaction:0 "Give Magic Eyes"
+ 
+ IS_SCOURGE:0 "You are the Scourge: $VALUE|=+0$"
```
Not part. It's visible for the Lich King.

NovelConcept commented on localization/english/wc_traits_l_english.yml:
```diff
@@ -487,6 +489,15 @@
+ TRAIT_FLAG_DESC_makes_incapable:0 "@warning_icon!#X Makes [wc_incapable|E]#!"
+
+ trait_lich_queen:0 "Lich Queen"
+ trait_lich_king_common_desc:0 "Originally created by Kil'jaeden from the spirit of the orc shaman Ner'zhul trapped inside the Helm of Domination. The Lich King's unimaginable necromantic potential allows him to control hundreds of thousands of walking undead."
```
```suggestion
 trait_lich_king_common_desc:0 "Originally created by Kil'jaeden from the spirit of the orc shaman, Ner'zhul, who he bound within the Helm of Domination. The Lich King's vast, telepathic influence and unimaginable necromantic power allows him to control endless hordes of living dead."
```

NovelConcept commented on localization/english/wc_traits_l_english.yml:
```diff
@@ -487,6 +489,15 @@
+ trait_lich_king_common_desc:0 "Originally created by Kil'jaeden from the spirit of the orc shaman Ner'zhul trapped inside the Helm of Domination. The Lich King's unimaginable necromantic potential allows him to control hundreds of thousands of walking undead."
+
+ trait_in_ice_prison:0 "Inside Ice Prison"
+ trait_in_ice_prison_desc:0 "A soul has been bound to the Helm of Domination, seated and trapped inside a frozen block of ice which would fittingly be named the Frozen Throne."
```
```suggestion
 trait_in_ice_prison_desc:0 "A soul has been bound to the Helm of Domination, seated and trapped within a diamond-hard block of ice, fittingly named the Frozen Throne."
```

NovelConcept commented on localization/english/wc_traits_l_english.yml:
```diff
@@ -487,6 +489,15 @@
+
+ trait_in_ice_prison:0 "Inside Ice Prison"
+ trait_in_ice_prison_desc:0 "A soul has been bound to the Helm of Domination, seated and trapped inside a frozen block of ice which would fittingly be named the Frozen Throne."
+ trait_in_ice_prison_character_desc:0 "[ROOT.GetCharacter.GetFirstNamePossessiveNoTooltip] soul has been bound to the Helm of Domination, seated and trapped inside a frozen block of ice which would fittingly be named the Frozen Throne."
```
```suggestion
 trait_in_ice_prison_character_desc:0 "[ROOT.GetCharacter.GetFirstNamePossessiveNoTooltip] soul has been bound to the Helm of Domination, seated and trapped within a diamond-hard block of ice, fittingly named the Frozen Throne."
```

ercarp commented on localization/english/wc_traits_l_english.yml:
```diff
@@ -487,6 +489,15 @@
+ TRAIT_FLAG_DESC_makes_incapable:0 "@warning_icon!#X Makes [wc_incapable|E]#!"
+
+ trait_lich_queen:0 "Lich Queen"
+ trait_lich_king_common_desc:0 "Originally created by Kil'jaeden from the spirit of the orc shaman, Ner'zhul, who he bound within the Helm of Domination. The Lich King's vast, telepathic influence and unimaginable necromantic power allows him to control endless hordes of living dead."
```
If there's localization for `trait_lich_queen`, shouldn't there also be a `trait_lich_queen_common_desc`? I'm only seeing the male version here.

zumbak04 replied:
Changed in bd21345209f4ec7c31efca92d215c3135a6fda3f.

ercarp reviewed (commented):
> The Necro government's Offer Vassalization has an option to turn Undead.

Works. ✔️

> The Scourge vassalizes easier. Especially vrykul who believe the Lich King is the death god.

I wonder if this should be added as an opinion modifier as well? It's a bit strange that all the vrykul have -100 opinion of me, but I can still offer vassalage to them because they "believe I'm the Death God".

If they believe my character to be the Death God, shouldn't they have a higher opinion of them? I'm not saying they should have a positive opinion of me necessarily, but just a slight increase so it's **1)** easier to tell who is going to accept my offer to vassalize them, and **2)** to offset the heavy -120 negative opinion penalty from "Mord'truar is Evil".

<details><summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/78575425/136656465-bb7180df-03d1-4e4c-877c-70d5f869e072.png)
![image](https://user-images.githubusercontent.com/78575425/136656469-184e5113-5a19-4a4a-a975-eb182758c3f5.png)

</details>

> Due to it, the Lich King can't marry and romance.

When I tried to seduce or romance someone, I noticed there's a missing bit of localization:

<details><summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/78575425/136656551-179258e0-0e49-43f2-ab9c-87f7c027eed5.png)
![image](https://user-images.githubusercontent.com/78575425/136656581-00d7a4b8-0944-4a4d-9bfb-188df3f0f329.png)

</details>

> Ner'zhul has the In Ice Prison trait. Due to it, Ner'zhul is incapable and can't lead armies.

Works. ✔️

> The Scourge spawns after the destruction of Draenor:
>- The Scourge gets event troops, money, high dread, piety, and prestige.
>- The Scourge has a CB and actively expands in Northrend.
>   - The Scourge CB vassalizes mortals and turns them undead.

Appears to work correctly. ✔️

> The Scourge doesn't trigger the Alliance to appear unless the Scourge is in the Eastern Kingdoms.
>
> The Lich King can't convert to faiths with banned Death Magic via the Faith GUI.

Not sure how to test these.

All of the localization looked good to me. Don't see any changes that would be needed. ✔️

zumbak04 commented:
> Not sure how to test these.

First, just confirm the Alliance doesn't create when the Scourge appears.

Second, you may convert to faith via the Faith Interface, there's a button.

ValianBlue reviewed (commented):
<hr>
> The Scourge has a CB and actively expands in Northrend. The Scourge CB vassalizes mortals and turns them undead.

_I did not see this option._

<details>
<summary></summary>

![image](https://user-images.githubusercontent.com/40177753/136669510-8a955397-be63-4290-aa39-afc15fa3de44.png)
</details>

<details>
<summary></summary>

![image](https://user-images.githubusercontent.com/40177753/136669491-047f609d-3d27-4136-be89-007e036ccb7a.png)
</details>
<hr>
> The Scourge doesn't trigger the Alliance to appear unless the Scourge is in the Eastern Kingdoms.

_I grabbed several pieces of EK and waited a few years but the Alliance did not spawn. Is this a bookmark specific thing?_

<details>
<summary></summary>

![image](https://user-images.githubusercontent.com/40177753/136669838-7ee4ec1f-f1fd-490e-a622-95a4f7bdf5eb.png)
</details>
<hr>
_Is the Lich King trait meant just for Ner'zhul, or will it be applied to Arthas and Bolvar (and whomever else) in the future? If it's the latter case, I would change the description of the trait to remove the lines mentioning Ner'zhul._

<details>
<summary></summary>

![image](https://user-images.githubusercontent.com/40177753/136669379-cb93de30-bc14-44e1-8562-58a7d0429d79.png)
</details>
<hr>
_Do we want the Scourge event troops to stand down on succession?_

<details>
<summary></summary>

![image](https://user-images.githubusercontent.com/40177753/136669371-f1917077-4a3d-4d1e-bf43-2b2dff2ef7a8.png)
</details>
<hr>
_Is it possible to make a character immune to diseases / have not them pop up? I understand that Undead get a Godlike boost to health, but it does ruin the immersion to have an event popup for having... a common cold._

<details>
<summary></summary>

![image](https://user-images.githubusercontent.com/40177753/136669776-9b9b9535-3318-4469-9555-8d238a3447d4.png)
</details>
<hr>

_All other features that I have not mentioned appear to be working correctly._

zumbak04 commented:
> _I did not see this option._

It's the Great Invasion on your second screenshot.

> _I grabbed several pieces of EK and waited a few years but the Alliance did not spawn. Is this a bookmark specific thing?_

Fixed in a9b87422ac9f94c78c4d07c2af9a6d5f4689ef60.

> _Is the Lich King trait meant just for Ner'zhul, or will it be applied to Arthas and Bolvar (and whomever else) in the future? If it's the latter case, I would change the description of the trait to remove the lines mentioning Ner'zhul._

It's the latter case.
First, it's a part of the history of the Helm. It's indeed Ner'zhul's soul trapped there.
Second, every Lich King is fused with Ner'zhul to become one person.

> _Do we want the Scourge event troops to stand down on succession?_

Changed in af4b56ee4976ccb8063122c02fe0ca42cc48f120.

> _Is it possible to make a character immune to diseases / have not them pop up? I understand that Undead get a Godlike boost to health, but it does ruin the immersion to have an event popup for having... a common cold._

It's on to-do list.

ValianBlue reviewed (approved):
>It's the Great Invasion on your second screenshot.

_Ah, I overlooked that. Is it possible to change the CB title to something like 'Scourge Invasion' to be more descriptive?_

>Fixed in a9b8742.

_Confirmed. ✅_ 

>It's the latter case.
First, it's a part of the history of the Helm. It's indeed Ner'zhul's soul trapped there.
Second, every Lich King is fused with Ner'zhul to become one person.

_I understand, I still would like the description to be changed to be more representative of that, but the current one works for now._

>Changed in af4b56e.

_Confirmed. ✅_

TheDrPibb1999 reviewed (approved):
All looks good to me, went through checklist, lk trait is awesome. good work!
